scripts/sysman/EXT_EngineActivity.rst

```diff
@@ -0,0 +1,76 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+    s=tags['$s']
+    S=s.upper()
+%>
+:orphan:
+
+.. _ZES_extension_engine_activity:
+
+==========================
+ Engine Activity Extension
+==========================
+
+API
+----
+
+* Functions
+
+    * ${s}EngineGetActivityExt
+
+* Structures
+
+    * ${s}_engine_ext_properties_t
+
+Engine Activity
+~~~~~~~~~~~~~~~~~~
+
+This extension provides capability to user to gather the activity stats for each Virtual Function (VF) associated with engine group.
+Intended use case of the feature is to query utilization stats from Physical Function (PF) interface. However, the extension
+may return error code $X_RESULT_ERROR_UNSUPPORTED_FEATURE if an environment does not support it.
+
+The following pseudo-code demonstrates a sequence for obtaining the engine activity for all Virtual Functions from Physical Function environment:
+
+.. parsed-literal::
+
+    // Query for engine properties and gather number of virtual function instances by passing extension structure.
+    // Number of VFs is per engine handle and equates to number of VFs actively using
+    // the resource represented by the engine handle.
+    // Together with pCount from GetActivityExt, user can gather the full view of engine
+    // utilization for all VFs and PF as shown below
+
+    ${s}_engine_properties_t engineProps;
+    ${s}_engine_ext_properties_t engineExtProps;
+    engineProps.stype = ZES_STRUCTURE_TYPE_ENGINE_PROPERTIES;
+    engineProps.pNext = &engineExtProps;
+    engineExtProps.stype = ZES_STRUCTURE_TYPE_ENGINE_EXT_PROPERTIES;
+
+    ${s}EngineGetProperties(hEngine, &engineProps);
+    uint32_t numVF = engineExtProps.countOfVirtualFunctionInstance;
+
+    // Query count of engine stats to gather
+    uint32_t engineStatCount = 0;
+    {s}EngineGetActivityExt(hEngine, &engineStatCount, nullptr);
+
+    // Gather engine stats for all VFs and PF.
+    // Gather two snapshots of engine activity and calculate percentage
+    // utilization per formula in ${s}_engine_stats_t
+    ${s}_engine_stats_t* engineStats0 = (${s}_engine_stats_t*) allocate(engineStatCount * sizeof(${s}_engine_stats_t));
+    {s}EngineGetActivityExt(hEngine, &engineStatCount, engineStats0);
+    sleep(1);
+    ${s}_engine_stats_t* engineStats1 = (${s}_engine_stats_t*) allocate(engineStatCount * sizeof(${s}_engine_stats_t));
+    {s}EngineGetActivityExt(hEngine, &engineStatCount, engineStats1);
+
+    for (uint32_t vf = 0; vf < engineStatCount; ++vf) {
+        // Ensure there was activity before calculation
+        if (engineStats0[vf].activeTime == 0 && engineStats1[vf].activeTime == 0) {
```

Comment:
Clarification: Do we need this check, since the utilization would show 0%.
Also since the active time is monotonically increasing, 0% utilization may mean that activeTime at 2 snapshots are same (may not be necessarily 0)

Reply:
yep, will remove check

Reply:
fixed